sup3r/preprocessing/exogenous_data_handling.py

```diff
@@ -0,0 +1,83 @@
+"""Sup3r exogenous data handling"""
+
+import logging
+import numpy as np
+
+from sup3r.utilities.topo import TopoExtract
+
+
+logger = logging.getLogger(__name__)
+
+
+class ExogenousDataHandler:
+    """Class to extract exogenous features for multistep forward passes. e.g.
+    Multiple topography arrays at different resolutions for multiple spatial
+    enhancement steps."""
+
+    def __init__(self, file_paths, features, source_h5, s_enhancements,
+                 agg_factors, target=None, shape=None, raster_file=None,
+                 max_delta=20, input_handler=None):
+        """
+        Parameters
+        ----------
+        file_paths : str | list
+            A single source h5 wind file to extract raster data from or a list
+            of netcdf files with identical grid. The string can be a unix-style
+            file path which will be passed through glob.glob
+        features : list
+            List of exogenous features to extract from source_h5
+        source_h5 : str
+            Filepath to source wtk or nsrdb file to get hi-res (2km or 4km)
+            data from which will be mapped to the enhanced grid of
+            the file_paths input
+        s_enhancements : list
+            List of factors by which the Sup3rGan model will enhance the
+            spatial dimensions of low resolution data from file_paths input.
+            For example, if file_paths has 100km data and s_enhance is 4, this
+            class will output a feature raster corresponding to the file_paths
+            grid enhanced 4x to ~25km
+        agg_factor : list
+            List of factors by which to aggregate the topo_source_h5 elevation
+            data to the resolution of the file_paths input enhanced by
+            s_enhance. For example, if file_paths has 100km data and s_enhance
+            is 4 resulting in a desired resolution of ~25km and topo_source_h5
+            has a resolution of 4km, the agg_factor should be 6 so that 6x 4km
+            cells are averaged to the ~25km enhanced grid.
```

Comment:
Hey also can you update this docstring here and in the topo extractor? The agg factor for this example should be 6 in either spatial axis, so the agg_factor should actually be 6**2=36